Tests/Indicators/StochasticTests.cs

```diff
@@ -21,47 +21,56 @@
 namespace QuantConnect.Tests.Indicators
 {
     [TestFixture]
-    public class StochasticTests
+    public class StochasticTests : CommonIndicatorTests<IBaseDataBar>
     {
-        [Test]
-        public void ComparesAgainstExternalDataOnStochasticsK()
+        protected override IndicatorBase<IBaseDataBar> CreateIndicator()
         {
-            var stochastics = new Stochastic("sto", 12, 3, 5);
+            return new Stochastic(12, 3, 5);
+        }
 
-            const double epsilon = 1e-3;
+        protected override string TestFileName => "spy_with_stoch12k3.txt";
 
-            TestHelper.TestIndicator(stochastics, "spy_with_stoch12k3.txt", "Stochastics 12 %K 3",
-                (ind, expected) => Assert.AreEqual(expected, (double) ((Stochastic) ind).StochK.Current.Value, epsilon)
-                );
-        }
+        protected override string TestColumnName => "%D 5";
+
+        protected override Action<IndicatorBase<IBaseDataBar>, double> Assertion =>
+            (indicator, expected) =>
+                Assert.AreEqual(expected, (double)((Stochastic)indicator).StochD.Current.Value, 1e-3);
 
         [Test]
-        public void PrimaryOutputIsFastStochProperty()
+        public void ComparesAgainstExternalDataOnStochasticsK()
         {
-            var stochastics = new Stochastic("sto", 12, 3, 5);
-
-            TestHelper.TestIndicator(stochastics, "spy_with_stoch12k3.txt", "Stochastics 12 %K 3",
-                (ind, expected) => Assert.AreEqual((double) ((Stochastic) ind).FastStoch.Current.Value, ind.Current.Value)
-                );
+            TestHelper.TestIndicator(
+                CreateIndicator(),
+                TestFileName,
+                "Stochastics 12 %K 3",
+                (ind, expected) => Assert.AreEqual(
+                    expected,
+                    (double) ((Stochastic) ind).StochK.Current.Value,
+                    1e-3
+                )
+            );
         }
 
         [Test]
-        public void ComparesAgainstExternalDataOnStochasticsD()
+        public void PrimaryOutputIsFastStochProperty()
         {
-            var stochastics = new Stochastic("sto", 12, 3, 5);
-
-            const double epsilon = 1e-3;
-            TestHelper.TestIndicator(stochastics, "spy_with_stoch12k3.txt", "%D 5",
-                (ind, expected) => Assert.AreEqual(expected, (double) ((Stochastic) ind).StochD.Current.Value, epsilon)
-                );
+            TestHelper.TestIndicator(
+                CreateIndicator(),
+                TestFileName,
+                "Stochastics 12 %K 3",
+                (ind, expected) => Assert.AreEqual(
+                    (double) ((Stochastic) ind).FastStoch.Current.Value,
+                    ind.Current.Value
+                )
+            );
         }
 
         [Test]
         public void ResetsProperly()
```

Comment:
Should this be an `override`?

Reply:
I didn't use override it so that the unit test can run `ResetsProperly` from this class and from the base.

Reply:
👍 it should be marked as `new` in that case to avoid a compiler warning:
`public new void ResetsProperly()`